src/dxtbx/format/FormatNXmxEigerFilewriter.py

```diff
@@ -49,15 +48,22 @@ def _get_nxmx(self, fh: h5py.File):
         if nxdetector.underload_value is None:
             nxdetector.underload_value = 0
 
-        # older firmware versions had the detector dimensions inverted
-        fw_version_string = (
-            fh["/entry/instrument/detector/detectorSpecific/eiger_fw_version"][()]
-            .decode()
-            .replace("release-", "")
-        )
-        if version.parse("2022.1.2") > version.parse(fw_version_string):
-            for module in nxdetector.modules:
+        # Some firmware versions had the detector dimensions swapped. The correct way
+        # is the number of pixels along the slow axis first, then the fast axis. Here
+        # swap any that look like they are (fast, slow) instead.
+        swapped_dims = {
```

Comment:
Would be tempted to define the known module dimensions at the top level of the file, rather than having the array inline i.e.

```python
KNOWN_MODULE_SWAPPED_DIMS = {
            (4148, 4362),
            (3108, 3262),
            (2068, 2162),
            (1028, 1062),
            (4148, 512),
            (2068, 512),
            (1028, 512),
}
```

etc.

Reply:
Then it is clear where new dimension have to go when some deep-pocketed beamline buys an Eiger 2 XE 25M 😆